### ISSUE TITLE
How to add data to a bar chart  dynamically

### ISSUE THREAD

**Issue by ppd324** [go-echarts/go-echarts#295]
I want to dynamically update the data in the bar chart，I want to dynamically update the data in the bar graph on the web page, but I haven't found a suitable function. Is it AddDataset?

**Comment by Koooooo-7** [comment]
Hi @ppd324 , there is no functions for this feat now.
I think you could add a custom js into the page and dynamic update the data as workaround.
Once the data change, it will re-render this page immediately.
FYI.  [echarts - dynamic load data](https://apache.github.io/echarts-handbook/en/how-to/data/dynamic-data/)

**Comment by beijJxy** [comment]
oh This function is so important, why not

**Comment by beijJxy** [comment]
> I want to dynamically update the data in the bar chart，I want to dynamically update the data in the bar graph on the web page, but I haven't found a suitable function. Is it AddDataset?

Did you solve the problem

**Comment by ppd324** [comment]
No, dynamically adding data should require JavaScript processing,Can draw it on the front end.

**Comment by Koooooo-7** [comment]
>  A: create js function to poll data from a file which is being updated by go program and add it to rendered HTML page 
> B: create a js server and send data updates from go with APIs/Socket connection 
> C: just call page.Render() again
> 
> Do you have any idea which option will give best performance/feasibility? what should I choose?

Hi @foobarchamp , I think the second option is more reasonable to me.
I suppose you could setup a websocket and do the options/data update when the new data/options comes.
Passing the new options/data and do the `<the chart instance>.setOption(newData)` to do the update dynamically.

OptionA is in a infinity polling is unnessary.
OptionC needs to reload the whole page every times.

**Comment by CanICodeDis** [comment]
I also have a question related to this. I cannot find a method to set the chart ID in go. I have multiple charts on a page and have to identify them in the js. But I cannot find out where to set the ID to each chart.
